scripts/_common.py

```diff
@@ -26,14 +27,33 @@ def get_secret_string(session, *, secret_id):
     return secrets.get_secret_value(SecretId=secret_id)["SecretString"]
 
 
-def get_api_es_client(session):
+def get_pipeline_es_client(session, *, pipeline_date, action, doc_type):
     """
     Returns an Elasticsearch client for the catalogue cluster.
     """
-    host = get_secret_string(session, secret_id="catalogue/ingestor/es_host")
-    port = get_secret_string(session, secret_id="catalogue/ingestor/es_port")
-    protocol = get_secret_string(session, secret_id="catalogue/ingestor/es_protocol")
-    username = get_secret_string(session, secret_id="catalogue/ingestor/es_username")
-    password = get_secret_string(session, secret_id="catalogue/ingestor/es_password")
+    secret_path = f"elasticsearch/pipeline_storage_{pipeline_date}"
+
+    if doc_type not in ["work", "image"]:
+        raise Exception("doc_type must be 'work' or 'image'")
+
+    if action == "read":
+        user = "read_only"
+    elif action == "write":
+        user = f"{doc_type}_ingestor"
```

Comment:
Should we create a new `scripts` user to avoid having to recycle more that 1 service if a key leaks?

Reply:
They're quite easy to add with the [script we use for this]https://github.com/wellcomecollection/catalogue-pipeline/blob/main/pipeline/terraform/scripts/create_pipeline_storage_users.py).